pkg/kubelet/cm/dra/state/cdi.go

```diff
@@ -23,19 +23,47 @@ limitations under the License.
 // Long term it would be good to avoid this duplication:
 // https://github.com/container-orchestrated-devices/container-device-interface/issues/97
 
-package dra
+package state
 
 import (
 	"errors"
 	"fmt"
 	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
 const (
 	// annotationPrefix is the prefix for CDI container annotation keys.
 	annotationPrefix = "cdi.k8s.io/"
 )
 
+// ContainerInfo contains information required by the runtime to consume prepared resources.
+type ContainerInfo struct {
+	// The Annotations for the container
+	Annotations []kubecontainer.Annotation
+}
+
+// generate container annotations using CDI UpdateAnnotations API.
+func generateCDIAnnotations(
+	claimUID types.UID,
+	driverName string,
+	cdiDevices []string,
+) ([]kubecontainer.Annotation, error) {
+	annotations, err := updateAnnotations(map[string]string{}, driverName, string(claimUID), cdiDevices)
+	if err != nil {
+		return nil, fmt.Errorf("can't generate CDI annotations: %+v", err)
+	}
+
+	kubeAnnotations := []kubecontainer.Annotation{}
+	for key, value := range annotations {
+		kubeAnnotations = append(kubeAnnotations, kubecontainer.Annotation{Name: key, Value: value})
+	}
+
+	return kubeAnnotations, nil
+}
+
```

Comment:
I'd suggest not to change this file as it contains a code copied from another project (see a comment at the top of the file).

Reply:
I see so I will move the generateCDIAnnotations and ContainerInfo to the state_checkpoint.go

Reply:
After thinking about it a little bit more, I agree that moving generateCDIAnnotations to cdi.go makes sense. When we stop supporting annotations we'll need to remove this function and its calls anyway, so having it in cdi.go is reasonalble.